packages/reports/addon/components/filter-builders/time-dimension.ts

```diff
@@ -307,4 +315,21 @@ export default class TimeDimensionFilterBuilder extends BaseFilterBuilderCompone
       values: arr(newInterval),
     });
   }
+
+  @action
+  updateParameters(key: string, value: string) {
+    // if the grain is changing, update the start/end values as well
+    if (key === 'grain') {
+      const internalOp = internalOperatorForValues(this.args.filter);
+      const values = valuesForOperator(this.args.filter, value as Grain, internalOp);
+      this.args.onUpdateFilter({
+        parameters: { ...this.args.filter.parameters, [key]: value },
+        values: values,
+      });
+    } else {
+      this.args.onUpdateFilter({
+        parameters: { ...this.args.filter.parameters, [key]: value },
+      });
```

Comment:
sort of - I'm not entirely sure what each of the specific names are, but I know that this eventually gets passed back to the "UPDATE_FILTER" report action. I pretty sure the reason it's specifically called this way is so that ember's automatic tracking recognizes what's being accessed, but it's the same as what's done in the [base dimension filter](https://github.com/yavin-dev/framework/blob/master/packages/reports/addon/components/filter-builders/base.ts#L78)